src/main/java/com/microsoft/spring/data/gremlin/conversion/MappingGremlinConverter.java

```diff
@@ -0,0 +1,70 @@
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See LICENSE in the project root for
+ * license information.
+ */
+package com.microsoft.spring.data.gremlin.conversion;
+
+import com.microsoft.spring.data.gremlin.mapping.GremlinPersistentEntity;
+import com.microsoft.spring.data.gremlin.mapping.GremlinPersistentProperty;
+import org.apache.commons.lang3.NotImplementedException;
+import org.springframework.context.ApplicationContext;
+import org.springframework.context.ApplicationContextAware;
+import org.springframework.core.convert.ConversionService;
+import org.springframework.core.convert.support.GenericConversionService;
+import org.springframework.data.convert.EntityConverter;
+import org.springframework.data.mapping.context.MappingContext;
+
+public class MappingGremlinConverter
+        implements EntityConverter<GremlinPersistentEntity<?>, GremlinPersistentProperty, Object, GremlinSource>,
+        ApplicationContextAware {
+
+    protected final MappingContext<? extends GremlinPersistentEntity<?>, GremlinPersistentProperty> mappingContext;
+    protected GenericConversionService conversionService;
+    private ApplicationContext applicationContext;
+
+    public MappingGremlinConverter(
+            MappingContext<? extends GremlinPersistentEntity<?>, GremlinPersistentProperty> context) {
+
+        this.mappingContext = context;
+        this.conversionService = new GenericConversionService();
+    }
+
+    public ApplicationContext getApplicationContext() {
+        return this.applicationContext;
+    }
+
+    @Override
+    public MappingContext<? extends GremlinPersistentEntity<?>, GremlinPersistentProperty> getMappingContext() {
+        return this.mappingContext;
+    }
+
+    @Override
+    public void setApplicationContext(ApplicationContext context) {
+        this.applicationContext = context;
+    }
+
+    @Override
+    public ConversionService getConversionService() {
+        return this.conversionService;
+    }
+
+    @Override
+    public <R extends Object> R read(Class<R> type, GremlinSource source) {
+        if (source == null) {
+            return null;
+        }
+
+        throw new NotImplementedException("read method of MappingGremlinConverter not implemented yet");
```

Comment:
If it's not implemented, shall we remove the `source == null` check here?

Reply:
Copy that, will add @NonNull for that.